02_bubble/srini/main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var outWriter io.Writer = os.Stdout
+
+func bubble(sliceToSort []int) []int {
+	sortedSlice := make([]int, len(sliceToSort))
+	copy(sortedSlice, sliceToSort)
+
+	lastIndex := len(sortedSlice) - 1
+
+	for i := 0; i < lastIndex; i++ {
+		swapped := true
```

Comment:
This should be initialised to `false` and set to true when you swap. The terminating condition should then be `!swapped`. As it is, the name is backwards to what is actually happening, which can be confusing.